pandas/tests/arrays/masked/test_arrow_compat.py

```diff
@@ -36,6 +37,27 @@ def test_arrow_roundtrip(data):
     tm.assert_frame_equal(result, df)
 
 
+def test_dataframe_from_arrow_types_mapper(any_int_ea_dtype):
+    int_dtype = pandas_dtype(any_int_ea_dtype)
+
+    def types_mapper(arrow_type):
+        if pa.types.is_boolean(arrow_type):
+            return pd.BooleanDtype()
+        elif pa.types.is_integer(arrow_type):
+            return int_dtype
```

Comment:
Is it actually needed to parametrize this over all int EA dtypes? Could also just use `pd.Int64Dtype()` like you use BooleanDtype.

Reply:
Don't you want to make sure `types_mapper` works for any integer dtype?

Reply:
Alternatively, I could have a separate column in the arrow record batch for each integer data type, but that seemed too much.

Reply:
> Don't you want to make sure types_mapper works for any integer dtype?

This is already implicitly tested in the `test_arrow_roundtrip` above (since that is parametrized for every dtype, and will under the hood make use of the same `dtype.__from_arrow__`)

But I am also fine with keeping the test as you have it now.

Reply:
Sounds good. I'll simplify, but add one column just so we test a code path that requires a copy because of differing data sizes.